dataretrieval/nwis.py

```diff
@@ -436,32 +438,57 @@ def _iv(**kwargs):
     """
-    Return a DataFrame containing all NWIS parameter codes.
+    Returns a DataFrame containing all NWIS parameter code information.
 
-    Note: NWIS may return incorrect column names. Rename them with
-
-    >>> df.rename(columns={key:value})
-
-    Parameters (Additional parameters, if supplied, will be used as query parameters).
+    Parameters 
     ----------
-        parameterCd: string or listlike
+        parameterCd: string or list
+            Accepts parameter codes or names
+        
+        partial: boolean
+            Default is True (partial querying). If False, the funciton will query only exact matches 
     Returns:
         DataFrame containing the USGS parameter codes and Metadata as tuple
     """
-    payload = {'radio_pm_search' : 'pm_search',
-               'pm_group' : 'All+--+include+all+parameter+groups',
-               'pm_search' : parameterCd,
-               'casrn_search' : None,
-               'srsname_search' : None,
-               'show' :  ['parameter_group_nm', 'casrn', 'srsname','parameter_units', 'parameter_nm'],
-               'format' : 'rdb'}
+    url = PARAMCODES_URL
+    payload = {'fmt':'rdb'}
+    
+    if parameterCd is None:
+        raise TypeError('The query must include a parameter name or code')
+        
+    else:
+        if isinstance(parameterCd, str): # when a single code or name is given
+            if parameterCd.lower() == "all": # Querying all parameters (this is also the default)
+                payload.update({'group_cd':'%'})
+                url = ALLPARAMCODES_URL
+            else: # this is for querying with a single parameter code or name
+                if partial:
+                    parameterCd ='%{0}%'.format(parameterCd)
+                payload.update({'parm_nm_cd':parameterCd})  
+        elif isinstance(parameterCd, list): # Querying with a list of parameters names, codes, or mixed
+            l = []
+            for param in parameterCd:
+                if isinstance(param, str):
+                    if partial:
+                        param ='%{0}%'.format(param)
+                    payload.update({'parm_nm_cd':param})
+                    response = query(url, payload)
+                    if len(response.text.splitlines()) < 10: # empty query
+                        raise TypeError('One of the parameter codes or names entered does not return any information,'\
+                                        ' please try a different value. Consider using partial = True to extend query results')
+                    l.append(_read_rdb(response.text))
+                else:
+                    raise TypeError('Parameter information (code or name) must be type string or list')
+            return pd.concat(l), _set_metadata(response)
+        else:
+            raise TypeError('Parameter information (code or name) must be type string or list')
     
-    payload.update(kwargs)
-    url = WATERDATA_URL + 'pmcodes/pmcodes'
```

Comment:
This url changed and the mock in the test needs to be updated with the new url, I believe it is here https://github.com/UCHIC/dataretrieval/blob/e8ce28b5982eaae74518b7f8f7241c589c3cf697/tests/waterservices_test.py#L285